### PR TITLE
Export multiple components from Renderer component

### PR DESCRIPTION
**Pull request by JSReds**
Export multiple components from Renderer component

Change-type: minor
Signed-off-by: Andrea Rosci <andrear@balena.io>

---
##### Contributor checklist

- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/components/Renderer/index.tsx [hunk 1]
```diff
@@ -155,21 +185,11 @@ const RendererBase = ({
 	});
 	const processedValue = getValue(value, schema, processedUiSchema);
 
-	if (processedValue === undefined || processedValue === null) {
-		return null;
-	}
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:160]: This seems to change the behavior.
Was the current behavior a bug?

File: src/components/Renderer/index.tsx [hunk 2]
```diff
@@ -70,6 +66,40 @@ export const getWidget = (
...
+
+export const Widget = ({
+	value,
+	schema,
+	extraFormats,
+	...otherProps
+}: WidgetProps) => {
+	const processedValue = getValue(value, schema);
+
+	if (processedValue === undefined || processedValue === null || !schema) {
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:86]: Let's simplify it a bit:
```suggestion
	if (processedValue == null || !schema) {
```

File: src/components/Renderer/index.tsx [hunk 3]
```diff
@@ -155,21 +185,11 @@ const RendererBase = ({
 	});
 	const processedValue = getValue(value, schema, processedUiSchema);
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:186]: I guess we now should be passing the plain `value` to the `Widget`, since the `Widget` is also doing the processing now, right?

File: src/components/Renderer/index.tsx [hunk 4]
```diff
@@ -70,6 +66,40 @@ export const getWidget = (
...
+export interface WidgetProps {
+	value?: Value;
+	schema?: JSONSchema;
+	extraContext?: object | undefined;
+	uiSchema?: UiSchema;
+	uiSchemaWidget?: UiSchema['ui:widget'];
+	extraFormats?: Format[];
+}
+
+export const Widget = ({
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:78]: Not a strong opinion, but now that we exporting it, how making this a bit more specific?
How about `ValueWidget` or `RendererWidget`?

File: src/extra/MarkdownEditor/defaultStyle.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ export default css`
 	}
 	.CodeMirror-gutter-filler,
 	.CodeMirror-scrollbar-filler {
-		background-color: transparent;
+		background-color: #fff;
```
Comment (thgreasi) [inline src/extra/MarkdownEditor/defaultStyle.ts:22]: Can we get some context on why these changes are needed?
Could they be in a separate PR?

File: src/index.ts [hunk 6]
```diff
@@ -27,7 +27,13 @@ export {
 	RenditionUiSchema,
 } from './components/Form';
 
-export { Renderer, RendererProps } from './components/Renderer';
+export {
+	Renderer,
+	RendererBase,
```
Comment (thgreasi) [inline src/index.ts:32]: Do we also need the RendererBase?
I mean if we only need the Widget for autoUI, then I would avoid also exporting any *Base components.

File: src/extra/MarkdownEditor/defaultStyle.ts (hunk 5 above)
Comment (JSReds) [reply src/extra/MarkdownEditor/defaultStyle.ts:22]: This is auto generated when running npm run generate-snapshots

File: src/index.ts (hunk 6 above)
Comment (JSReds) [reply src/index.ts:32]: Sorry this remained from tests. Removing it

File: src/components/Renderer/index.tsx (hunk 1 above)
Comment (JSReds) [reply src/components/Renderer/index.tsx:160]: no you right, changed to value == null

File: src/extra/MarkdownEditor/defaultStyle.ts (hunk 5 above)
Comment (thgreasi) [reply src/extra/MarkdownEditor/defaultStyle.ts:22]: @JSReds let's then have it as the first commit on its own just to be clear.

**Comment by LucianBuzzo** [comment]
@JSReds Why is this change needed? We're breaking an abstraction layer by exposing the internals of this component, making it harder to change in the future (we can't modify the implementation without it being a breaking change).

File: src/components/Renderer/index.tsx [hunk 7]
```diff
@@ -153,7 +184,8 @@ const RendererBase = ({
 		uiSchema,
 		extraContext,
 	});
-	const processedValue = getValue(value, schema, processedUiSchema);
+
+	const processedValue = getValue(value, schema, uiSchema);
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:188]: Is it equivalent that we are now passing `uiSchema` instead of `processedUiSchema`?

**Review by thgreasi — COMMENTED** [review]
Other than that LGTM :+1: 
Feel free to self-certify

**Comment by JSReds** [comment]
@LucianBuzzo, The Renderer component is very heavy as an export, as it carries asRenditon with it and has validation checks ( that in multiple cases are completely skipped). By dividing the component in this way, we would be able in an autoUI context to bring with us only the Widget component which is what we need in a cell context (cell of a table). I was splitting the component in order to avoid any duplication in the code. But if this is inconvenient for others, we can safely create another component. What do you think ?

**Comment by LucianBuzzo** [comment]
@JSReds If the intention is to re-use the individual widget fields in multiple places then the widgets should probably be a separate component, though this can happen as a follow-up PR if its deemed necessary.

**Comment by JSReds** [comment]
Closing in favour of a PR to come as AutoUI will be inside Rendition and will have the chance to use the component internally
